### PR TITLE
Reschedule CollectionSource occasionally 

### PR DESCRIPTION
**Pull request by Bathtor**
Reschedule CollectionSource occasionally (every 500k elements) so it doesn't execute a single handler forever. 

Kompact doesn't deal well with handlers hanging on to their threads forever. In particular not for the `ControlEvent` handler, since there are internal things that also need to happen in response to these.

### CODE REVIEW

**Comment by Max-Meldrum** [comment]
Looks good! Might as well remove the commented code though :)

File: execution-plane/arcon/src/stream/source/collection.rs [hunk 1]
```diff
@@ -52,21 +72,35 @@ where
...
+where
+    IN: ArconType,
+    OUT: ArconType,
+{
+    fn handle(&mut self, _event: Never) {
+        unreachable!("Never type has no instance");
+    }
+}
+
+//ignore_indications!(LoopbackPort, CollectionSource);
```
Comment (Bathtor) [inline execution-plane/arcon/src/stream/source/collection.rs:102]: Any idea why that doesn't work on Arcon? You pointing to an old Kompact version where it wasn't available, yet?

Comment (Max-Meldrum) [reply execution-plane/arcon/src/stream/source/collection.rs:102]: Was it added recently? Could be that we need to do a cargo update (as we are currently using git master as dependency)

Comment (Bathtor) [reply execution-plane/arcon/src/stream/source/collection.rs:102]: I mean...a few weeks ago...I don't know when you did cargo update last.

Anyway, if you do it at some point, try of that works, maybe and if yes delete the other variant. I can't remember if I once again forgot about supporting components with those pesky generic type parameters :D

**Comment by Bathtor** [comment]
Oehm...that didn't fail before. Flaky test?

**Comment by Max-Meldrum** [comment]
> Oehm...that didn't fail before. Flaky test?

flaky. Will add an issue to fix socket related tests

**Pull request merged**